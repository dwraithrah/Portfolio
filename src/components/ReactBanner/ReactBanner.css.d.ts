export const reactBanner: string;
export const reactBannerLogo: string;
export const appLogoSpin: string;
export const reactBannerHeader: string;
export const reactBannerIntro: string;
