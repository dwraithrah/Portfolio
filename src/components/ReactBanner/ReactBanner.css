.React-Banner {
  text-align: center;
}

.React-Banner-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.React-Banner-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.React-Banner-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

@media (max-width: 600px) {
  .React-Banner button {
    max-width: 110px;
  }
}
