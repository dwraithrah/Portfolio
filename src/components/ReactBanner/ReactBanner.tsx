import * as React from 'react';
import * as styles from './ReactBanner.css';
// import Grid from 'material-ui/Grid';
const logo = require('./logo.svg');

interface ReactBannerProps {
}

const ReactBanner: React.SFC<ReactBannerProps> = (props) => {
  return(
    <div className={styles.reactBanner}>
      <div className={styles.reactBannerHeader}>
        <img src={logo} className={styles.reactBannerLogo} alt="logo" />
        <h2>Welcome to React</h2>
      </div>
      <p className={styles.reactBannerIntro}>
        To get started, edit <code>src/App.tsx</code> and save to reload.
      </p>
    </div>
  );
};

export default ReactBanner;
