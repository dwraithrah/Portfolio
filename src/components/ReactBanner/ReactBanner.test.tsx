import * as React from 'react';
import ReactBanner from './ReactBanner';
import { shallow } from 'enzyme';

describe('<ReactBanner />', () => {
  const reactBanner = shallow(<ReactBanner />);
  it('should render without crashing', () => {
    expect(reactBanner).toMatchSnapshot();
  });
});
