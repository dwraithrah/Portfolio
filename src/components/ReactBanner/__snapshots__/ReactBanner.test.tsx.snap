// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactBanner /> should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReactBanner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <withStyles(GridWrapper)
        item={true}
        xs={12}
>
        <div
                className={undefined}
        >
                <img
                        alt="logo"
                        className={undefined}
                        src="logo.svg"
                />
                <h2>
                        Welcome to React
                </h2>
        </div>
        <p
                className={undefined}
        >
                To get started, edit 
                <code>
                        src/App.tsx
                </code>
                 and save to reload.
        </p>
</withStyles(GridWrapper)>,
      "className": undefined,
      "container": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <img
                        alt="logo"
                        className={undefined}
                        src="logo.svg"
            />
            <h2>
                        Welcome to React
            </h2>
</div>,
          <p
            className={undefined}
>
            To get started, edit 
            <code>
                        src/App.tsx
            </code>
             and save to reload.
</p>,
        ],
        "item": true,
        "xs": 12,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className={undefined}
                src="logo.svg"
/>,
              <h2>
                Welcome to React
</h2>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": undefined,
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to React",
              },
              "ref": null,
              "rendered": "Welcome to React",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "To get started, edit ",
              <code>
                src/App.tsx
</code>,
              " and save to reload.",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            "To get started, edit ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.tsx",
              },
              "ref": null,
              "rendered": "src/App.tsx",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withStyles(GridWrapper)
          item={true}
          xs={12}
>
          <div
                    className={undefined}
          >
                    <img
                              alt="logo"
                              className={undefined}
                              src="logo.svg"
                    />
                    <h2>
                              Welcome to React
                    </h2>
          </div>
          <p
                    className={undefined}
          >
                    To get started, edit 
                    <code>
                              src/App.tsx
                    </code>
                     and save to reload.
          </p>
</withStyles(GridWrapper)>,
        "className": undefined,
        "container": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
>
              <img
                            alt="logo"
                            className={undefined}
                            src="logo.svg"
              />
              <h2>
                            Welcome to React
              </h2>
</div>,
            <p
              className={undefined}
>
              To get started, edit 
              <code>
                            src/App.tsx
              </code>
               and save to reload.
</p>,
          ],
          "item": true,
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className={undefined}
                  src="logo.svg"
/>,
                <h2>
                  Welcome to React
</h2>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": undefined,
                  "src": "logo.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome to React",
                },
                "ref": null,
                "rendered": "Welcome to React",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "To get started, edit ",
                <code>
                  src/App.tsx
</code>,
                " and save to reload.",
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              "To get started, edit ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "src/App.tsx",
                },
                "ref": null,
                "rendered": "src/App.tsx",
                "type": "code",
              },
              " and save to reload.",
            ],
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
